scipy/optimize/tests/_cython_examples/meson.build

```diff
@@ -0,0 +1,18 @@
+project('random-build-examples', 'c', 'cpp', 'cython')
```

Comment:
You're requiring a C++ compiler here but don't use it then. It may actually be nice to test with C and with C++ codegen separately, since the behavior may differ. If it's done only for one small `.pyx` file per scipy submodule, this seems doable without incurring must test runtime overhead or maintenance cost. WDYT?

Reply:
That's a good idea. I added this to each test by copying the existing `.pyx` at build time and specifying C++ as the language for Cython transpilation.